src/solvers/sat/satcheck_minisat2.cpp

```diff
@@ -140,8 +140,10 @@ void satcheck_minisat2_baset<T>::lcnf(const bvt &bv)
 
     solver->addClause_(c);
 
-    with_solver_hardness(
-      [&bv](solver_hardnesst &hardness) { hardness.register_clause(bv); });
+    size_t solver_clause_num = clause_counter;
```

Comment:
Assignment seems redundant, just use `clause_counter` in line 145.

Reply:
`clause_counter` is an inherited data member of `satcheck_minisat2_baset<T>` from `cnf_solvert` so I cannot pass it directly to the lambda capture list (not in scope). Will need to pass `this` instead. Thought it would be better to just pass a copy.